src/library/scala/collection/immutable/Set.scala

```diff
@@ -94,14 +95,31 @@ object Set extends IterableFactory[Set] {
 
   def empty[A]: Set[A] = EmptySet.asInstanceOf[Set[A]]
 
+  // Miscellaneous inaccessible internal collections which can be returned directly in `from`.
+  // Fails silently but a unit test asserts that the classes were found.
+  @nowarn private val exclusions =
+    List(
+      "scala.collection.immutable.MapOps$ImmutableKeySet",
+      "scala.collection.immutable.HashMap$HashKeySet",
+    ).map(name => try Class.forName(name): AnyRef catch _ => null)
```

Comment:
Oh, I had not seen this before. `Class.forName` doesn't work on Scala.js (and probably not on Scala Native either). Is there any reason not to make those classes `private[immutable]` so that they would actually be accessible?

Reply:
I must have been reluctant to widen access. But as I'm in for a penny, I'll update it.

Reply:
I removed the kludge, but I could not figure out how to express `instanceof` for the keyset of much-parameterized `MapOps`. That case was covered in a test: the test and the hypothetical LOC in `from` are commented out.